src/wallet/wallet.cpp

```diff
@@ -1710,7 +1710,8 @@ CAmount CWallet::loopTxsBalance(std::function<void(const uint256&, const CWallet
             method(it.first, it.second, nTotal);
         }
     }
-    return nTotal;
+    // Amounts can be negative (method could subtract them), but final balance shouldn't.
+    return std::max((CAmount) 0, nTotal);
```

Comment:
Hmm, I'm not really in favor of this. It could end up hiding other issues in the future. `loopTxsBalance` method should be an abstract loop over the txs only, abstracted from what could happen inside the method.

 Better to have it only in the method that needs it.

Reply:
Balances should never be negative, but I can see the point. Moved.